### DISCUSSION TITLE
Example of how to pull out a parameter?

### DISCUSSION THREAD

**Discussion by jaysonng** [netreconlab/parse-server-swift#42] [Q&A] [answered]
Quick help pls.

Maybe something simple I can't figure out.

How do I pull out a string I send in as a parameter "name" into the cloud code function Im testing with.

In the readme file example 

```
// A simple Parse Hook Function route that returns "Hello World".
app.post("hello",
         name: "hello") { req async throws -> ParseHookResponse<String> in
    // Note that `ParseHookResponse<String>` means a "successfull"
    // response will return a "String" type.
    if let error: ParseHookResponse<String> = checkHeaders(req) {
        return error
    }
    var parseRequest = try req.content
        .decode(ParseHookFunctionRequest<User, FooParameters>.self)
    
    // If a User made the request, fetch the complete user.
    if parseRequest.user != nil {
        parseRequest = try await parseRequest.hydrateUser(request: req)
    }
    
    // To query using the User's credentials who called this function,
    // use the options() method from the parseRequest
    let options = try parseRequest.options(req)
    let scores = try await GameScore.query.findAll(options: options)
    req.logger.info("Scores this user can access: \(scores)")
    return ParseHookResponse(success: "Hello world!")
}
```

If I have a String "name" passed as a parameter, how do I pull it out of the parseRequest object and FooParameters? 

When I do a dump(parseRequest), the information is nowhere to be seen but if I do a dump(req.content) I do see my "name" parameter there with the expected string.

basically, how do I pull out a parameter from a decoded FooParameter?

thanks,

**Comment by jaysonng** [comment]
Nevermind. I think I misunderstood the instructions in the readme file. 

I thought for some reason that FooParameters would somehow have the params I send. 

I just realized that the instructions meant I need to create a new parameter file and place it in the Parameters folder and that FooParameters is the example Parameter file. 

thanks.

**Reply by cbaker6** [reply, marked as answer]
> I just realized that the instructions meant I need to create a new parameter file and place it in the Parameters folder and that FooParameters is the example Parameter file.

Sounds like you figured it out, I used `FooParameters` as an example, but you build all of your parameters in that type and they will be decoded and available in your Function.

**Reply by jaysonng** [reply]
Parameters should be optionals as well correct? 

```
struct FooParameters: ParseHookParametable {
    var name: String?
 }
```

**Reply by cbaker6** [reply]
If you don’t require the parameter to be passed when calling the function, it can be optional, or else it is required or you pass a default value like: https://github.com/netreconlab/Parse-Swift/blob/47548b39f7c8e26cefc544718f29c408c4fd99db/Tests/ParseSwiftTests/ParseHookFunctionRequestTests.swift#L19-L21
